app/common/elasticsearchclient.py

```diff
@@ -243,16 +243,24 @@ def best_hit_search(self,
         #there are len(params.q) responses - one per query
         for i in range(len(results['responses'])):
             name = params.q[i]  #even though we are guaranteed that responses come back in order, and can match query to the result - this might be convenient to have       
+            #print "name =" + name 
             lower_name = name.lower()
             res = results['responses'][i]
             exact_match = False
             if(res['hits']['total']>0):
                 hit = res['hits']['hits'][0] #expect either 1 result or none
                 highlight = ''
+                fields = kwargs['fields'];
                 if 'highlight' in hit:
                     highlight = hit['highlight']
-                if(lower_name == hit['_source']['approved_symbol'].lower() or lower_name == hit['_id'].lower):
+                if (lower_name == hit['_id'].lower):
                     exact_match = True
+                elif 'search-object-target' in doc_filter:
```

Comment:
Is this logic correct?
both targets and diseases have _id
both targets and diseases have _type
shouldn't we try the id and depending on the type approved_symbol or full_name?
```
type_ = hit['_type']
id_ = hit['_id'].lower()
if lower_name == _id:
    exact_match = True
if (type_ == 'search-object-target'):
    if lower_name == hit['_source']['approved_symbol'].lower():
        exact_match = True
elif (type_ == 'search-object-disease'):
    if lower_name == hit['_source']['full_name']:
        exact_match = True
```